packages/pagination/src/Pagination.js

```diff
@@ -93,14 +96,21 @@ const Pagination = ({
     toggleLoading(false);
   };
 
-  useEffect(() => {
-    getPageData();
-  }, [
-    currentPage,
-    itemsPerPage,
-    isFunction(theItems) ? null : theItems,
-    ...watchList,
-  ]);
+  useDebounce(
+    () => {
+      if (shouldGetPageData) {
+        getPageData();
+      }
+    },
+    debounceTimeout,
```

Comment:
If `debounceTimeout` is now a variable, do we want to add that to the dependency array?